iceoryx_hoofs/include/iceoryx_hoofs/cxx/type_traits.hpp

```diff
@@ -55,17 +55,17 @@ using add_const_conditionally_t = typename add_const_conditionally<T, C>::type;
 /// @endcode
 ///
 template <typename>
-constexpr bool always_false_v = false;
-
+constexpr bool always_false_v{false};
 // windows defines __cplusplus as 199711L
-#if __cplusplus < 201703L && !defined(_WIN32)
+// AXIVION DISABLE STYLE AutosarC++19_03-A16.0.1: TODO
```

Comment:
The better approach would be to put this into the iceoryx platform and remove the axivion suppression.

In every platform under `iceoryx_hoofs/platform/{linux,mac,qnx,unix,win}/include/platform_settings.hpp` you find platform specific settings and adjustments. I would suggest to add there the following alias for windows, mac and unix
```cxx
template<typename C, typename... Cargs>
using invoke_result = std::invoke_result<C, Cargs...>;
```

For qnx and linux I would add:
```cxx
template <typename C, typename... Cargs>
using invoke_result = std::result_of<C(Cargs...)>;
```

With this we have to adjust the used `invoke_result` here to `platform::invoke_result`. Then we can remove the whole macro and have a very clean solution.

Reply:
@elfenpiff That's a very good suggested approach. have implemented it. Thanks.